toolchain/check/handle_export.cpp

```diff
@@ -62,7 +62,7 @@ auto HandleParseNode(Context& context, Parse::ExportDeclId node_id) -> bool {
     context.emitter()
         .Build(node_id, ExportRedundant)
         // Use the location of the export itself, not the exported instruction.
-        .Note(context.insts().GetLocId(inst_id), ExportPrevious)
+        .Note(context.insts().GetResolvedLocId(inst_id), ExportPrevious)
```

Comment:
This shouldn't need the explicit call, should it?
```suggestion
        .Note(SemIR::LocId(inst_id), ExportPrevious)
```

Reply:
The comment above says why this is here, but it's not very obvious: `// Use the location of the export itself, not the exported instruction.`

The reason is in the GetAbsoluteNodeIdImpl function, which is called eventually from ConvertLoc, it does this:
```
    if (auto bind_ref = cursor_inst.TryAs<ExportDecl>();
        bind_ref && bind_ref->value_id.has_value()) {
      cursor_inst_id = bind_ref->value_id;
      continue;
    }
```

So if the LocId is an ExportDecl InstId, then it  chases through it. But we want the location of the export itself. And to do that, it does so by getting the canonical location first. That causes the GetAbsoluteNodeIdImpl path to not see the export instruction. It's quite magical really. But that's why it was doing GetLocId() here before instead of passing `inst_id` even when we had SemIRLoc.

I will add more to the comment.